src/main/java/seedu/address/model/application/Application.java

```diff
@@ -79,6 +81,24 @@ public Details getDetails() {
         return details;
     }
 
+    public Optional<Tag> getPriorityTag() {
+        for (Tag tag: tags) {
+            if (tag.tagType.equals(TagType.PRIORITY)) {
+                return Optional.of(tag);
+            }
+        }
+        return Optional.empty();
+    }
+
+    public Optional<Tag> getApplicationStatusTag() {
+        for (Tag tag: tags) {
+            if (tag.tagType.equals(TagType.APPLICATION_STATUS)) {
+                return Optional.of(tag);
+            }
+        }
+        return Optional.empty();
+    }
+
```

Comment:
Like your use of Optional here. Also good use of the TagType to look through the Set<Tag> tags!